frontend/src/lib/terminal.js

```diff
@@ -0,0 +1,405 @@
+  const connectPromise = new Promise((resolve, reject) => {
+    const openHandler = () => {
+      ws.removeEventListener('open', openHandler)
+      ws.removeEventListener('error', errorHandler)
+
+      resolve()
+    }
+    const errorHandler = error => {
+      ws.removeEventListener('open', openHandler)
+      ws.removeEventListener('error', errorHandler)
+
+      reject(error)
+    }
+
+    ws.addEventListener('open', openHandler)
+    ws.addEventListener('error', errorHandler)
+  })
+
+  const connectTimeoutSeconds = 5
+  await pTimeout(connectPromise, connectTimeoutSeconds * 1000, `Could not connect within ${connectTimeoutSeconds} seconds`)
+
+  const podRunningPromise = new Promise((resolve, reject) => {
+    const resolveOnce = value => {
+      ws.removeEventListener('message', messageHandler)
+      ws.removeEventListener('close', closeHandler)
+
+      resolve(value)
+    }
+    const rejectOnce = reason => {
+      ws.removeEventListener('message', messageHandler)
+      ws.removeEventListener('close', closeHandler)
+
+      reject(reason)
+    }
+
+    const closeHandler = error => {
+      rejectOnce(error)
+    }
+    const messageHandler = ({ data: message }) => {
+      let event
+      try {
+        event = JSON.parse(message)
+      } catch (error) {
+        console.error('could not parse message')
+        return
+      }
+      const pod = event.object
+      if (typeof handleEvent === 'function') {
+        try {
+          handleEvent(event)
+        } catch (error) {
+          console.error('error during handleEvent', error.message)
+        }
+      }
+
+      const phase = get(pod, 'status.phase')
+      if (includes(['Failed', 'Succeeded'], phase)) {
+        rejectOnce(new Error(`Pod is in phase ${phase}`))
+        return
+      } else if (event.type === 'DELETED') {
+        rejectOnce(new Error('Pod deleted'))
+        return
+      }
+
+      const containerStatuses = get(pod, 'status.containerStatuses')
+      const terminalContainerStatus = find(containerStatuses, ['name', containerName])
+      const isContainerReady = get(terminalContainerStatus, 'ready', false)
+
+      if (phase === 'Running' && isContainerReady) {
+        resolveOnce()
+      }
+    }
+    ws.addEventListener('message', messageHandler)
+    ws.addEventListener('close', closeHandler)
+  })
+  return pTimeout(podRunningPromise, timeoutSeconds * 1000, `Timed out after ${timeoutSeconds}s`)
+}
+
+function getDetailedConnectionStateText (terminalContainerStatus) {
+  const state = get(terminalContainerStatus, 'state')
+  const stateKeys = intersection(['waiting', 'running', 'terminated'], keys(state))
+  const stateType = head(stateKeys)
+
+  let text = ''
+  if (!stateType) {
+    return text
+  }
+
+  text = `Container is ${stateType}`
+
+  const reason = get(terminalContainerStatus, ['state', stateType, 'reason'])
+  if (!reason) {
+    return text
+  }
+
+  return `${text}: ${reason}`
+}
+
+module.exports = { TerminalSession }
```

Comment:
`export class TerminalSession`
and change require to import..

fix also in other files in the frontend